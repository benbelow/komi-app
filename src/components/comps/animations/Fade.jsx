// @flow
import * as React from 'react'
import { Transition } from 'react-transition-group'
import styled from 'styled-components'

const delay = 100

export const FadeTransition = ({
  children,
  enter = delay,
  exit = delay,
  appear,
  ...props
}: {
  children: Function,
  enter?: number,
  exit?: number,
  appear?: boolean,
  props?: {}
}) => (
  <Transition
    exit
    appear={appear}
    {...props}
    timeout={{
      enter,
      exit,
    }}
  >
    {(status: string): React.Node => children(status)}
  </Transition>
)

export const FadeComp = styled.div`
  transition: all 100ms ease-in-out;

  transform: scale(
    ${({ status, hide }: { status: string, hide: boolean }) => {
    if (hide === true) {
      return 0.2
    }
    if (status === 'entering') {
      return 0.2
    }
    if (status === 'entered') {
      return 1
    }
    if (status === 'exiting') {
      return 0.2
    }
    if (status === 'exited') {
      return 0.2
    }
    return 1
  }}
  );
  opacity: ${({ status, hide }: { status: string, hide: boolean }) => {
    if (hide === true) {
      return 0
    }
    if (status === 'entering') {
      return 0
    }
    if (status === 'entered') {
      return 1
    }
    if (status === 'exiting') {
      return 0
    }
    if (status === 'exited') {
      return 0
    }
    return 1
  }};
  visibility: ${({ status, hide }: { status: string, hide: boolean }) => {
    if (hide === true) {
      return 'hidden'
    }
    if (status === 'entering') {
      return 'hidden'
    }
    if (status === 'entered') {
      return 'visible'
    }
    if (status === 'exiting') {
      return 'visible'
    }
    if (status === 'exited') {
      return 'hidden'
    }
    return 'visible'
  }};
`
