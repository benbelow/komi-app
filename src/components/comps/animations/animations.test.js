import 'jest-styled-components'
import React from 'react'
import renderer from 'react-test-renderer'
import { FadeTransition, FadeComp } from './Fade'

describe('"Fade" animation test suite', () => {
  test('"FadeComp" should match snapshot', () => {
    const comp = renderer.create(<FadeComp />)
    const tree = comp.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('"FadeTransition" should match snapshot', () => {
    const comp = renderer.create(<FadeTransition>{status => <FadeComp status={status} />}</FadeTransition>)
    const tree = comp.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
