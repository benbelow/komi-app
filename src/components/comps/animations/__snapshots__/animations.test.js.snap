// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"Fade" animation test suite "FadeComp" should match snapshot 1`] = `
.c0 {
  -webkit-transition: all 100ms ease-in-out;
  transition: all 100ms ease-in-out;
  -webkit-transform: scale( 1 );
  -ms-transform: scale( 1 );
  transform: scale( 1 );
  opacity: 1;
}

<div
  className="c0"
/>
`;

exports[`"Fade" animation test suite "FadeTransition" should match snapshot 1`] = `
.c0 {
  -webkit-transition: all 100ms ease-in-out;
  transition: all 100ms ease-in-out;
  -webkit-transform: scale( 0.2 );
  -ms-transform: scale( 0.2 );
  transform: scale( 0.2 );
  opacity: 0;
}

<div
  className="c0"
/>
`;
